### PR TITLE
#122 Add Hackathon endpoints

### PR DESCRIPTION
**Pull request by IvanObreshkov**
#122 Create logic for not allowing duplicate team members in the DB by checking if the email is already present in the DB

### CODE REVIEW

File: packages/api/configs/env.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ func GenerateToken(lengthOfToken int) string {
 }
 
 func ReturnAuthToken() string {
+	println(token)
```
Comment (asynchroza) [inline packages/api/configs/env.go:34]: Remove this.

File: packages/api/controllers/hackathon_team_controller.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	if err != nil {
+		return c.Status(http.StatusInternalServerError).JSON(responses.MemberResponse{Status: http.StatusInternalServerError, Message: "error", Data: &fiber.Map{"data": err.Error()}})
+	}
+
+	return c.Status(http.StatusCreated).JSON(responses.MemberResponse{Status: http.StatusCreated, Message: "success", Data: &fiber.Map{"data": result}})
+}
+
+func GetHackathonTeams(c *fiber.Ctx) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+
```
Comment (asynchroza) [inline packages/api/controllers/hackathon_team_controller.go:50]: Missing auth

**Review by asynchroza — CHANGES_REQUESTED** [review]
Busy to review this but from a quick look I see that the auth is missing in a lot of places. This is purely internal info so nobody should have access to that. @nikolayninov and team, please verify all endpoints are working as expected. This is a really big PR which should be handled carefully.

File: packages/api/controllers/hackathon_team_members_controller.go [hunk 3]
```diff
@@ -61,13 +62,26 @@ func CreateHackathonMember(c *fiber.Ctx) error {
 		SponsorShare:          member.SponsorShare,
 		NewsLetter:            member.NewsLetter}
 
+	cursor, err := teamMembersCollection.Find(
+		ctx,
+		bson.D{{"email", newHackathonTeamMember.Email}},
+	)
+	var results []models.TeamMember
+	if err = cursor.All(ctx, &results); err != nil {
```
Comment (asynchroza) [inline packages/api/controllers/hackathon_team_members_controller.go:70]: pull the error check on the line below

File: packages/api/controllers/hackathon_team_members_controller.go [hunk 4]
```diff
@@ -61,13 +62,26 @@ func CreateHackathonMember(c *fiber.Ctx) error {
...
+		return c.Status(http.StatusInternalServerError).JSON(responses.MemberResponse{Status: http.StatusInternalServerError, Message: "error", Data: &fiber.Map{"data": err.Error()}})
+	}
+	if len(results) > 1 {
```
Comment (asynchroza) [inline packages/api/controllers/hackathon_team_members_controller.go:73]: len >= 1

File: packages/api/controllers/hackathon_team_members_controller.go (hunk 3 above)
Comment (asynchroza) [inline packages/api/controllers/hackathon_team_members_controller.go:70]: Why didn't you implement this? https://www.mongodb.com/docs/manual/core/index-unique/

**Review by asynchroza — CHANGES_REQUESTED** [review]

File: packages/api/controllers/hackathon_team_members_controller.go (hunk 3 above)
Comment (IvanObreshkov) [reply packages/api/controllers/hackathon_team_members_controller.go:70]: I tried with a search index but I wasn't happy with the results. If I have to I would change it.

Comment (asynchroza) [reply packages/api/controllers/hackathon_team_members_controller.go:70]: Can you elaborate? What were the results?

File: packages/api/configs/env.go (hunk 1 above)
Comment (IvanObreshkov) [reply packages/api/configs/env.go]: Done

File: packages/api/controllers/hackathon_team_controller.go (hunk 2 above)
Comment (IvanObreshkov) [reply packages/api/controllers/hackathon_team_controller.go:50]: Done, also added auth to the other GET requests

File: packages/api/controllers/hackathon_team_members_controller.go (hunk 3 above)
Comment (IvanObreshkov) [reply packages/api/controllers/hackathon_team_members_controller.go:70]: This is [part ](https://www.mongodb.com/docs/drivers/go/current/fundamentals/crud/read-operations/text/ )of the Mongo DB documentation I looked at. And it was getting all documents in the DB. Now I am going to try and make it with unique index.

Comment (IvanObreshkov) [reply packages/api/controllers/hackathon_team_members_controller.go:75]: Done

File: packages/api/controllers/hackathon_team_members_controller.go (hunk 4 above)
Comment (IvanObreshkov) [reply packages/api/controllers/hackathon_team_members_controller.go]: Done

**Review by nikolayninov — DISMISSED** [review]

**Review by asynchroza — APPROVED** [review]

File: packages/api/.vscode/launch.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (asynchroza) [inline packages/api/.vscode/launch.json:1]: Do you use this task? If not remove it.

Comment (IvanObreshkov) [reply packages/api/.vscode/launch.json:1]: It is code completion

Comment (asynchroza) [reply packages/api/.vscode/launch.json:1]: It is not. It's a launching task. Remove it.

Comment (asynchroza) [reply packages/api/.vscode/launch.json:1]: It is not. Remove the .vscode folder under api dir.

Comment (IvanObreshkov) [reply packages/api/.vscode/launch.json]: Done

**Review by nikolayninov — APPROVED** [review]

**Comment by nikolayninov** [comment]
Can I merge now? @asynchroza :**

**Comment by asynchroza** [comment]
> Can I merge now? @asynchroza :**

Yes, but make sure you leave the part of the merge comment which states who are the co-authors of the PR. And the correct title!

**Comment by nikolayninov** [comment]
> > Can I merge now? @asynchroza :**
> 
> Yes, but make sure you leave the part of the merge comment which states who are the co-authors of the PR. And the correct title!

What do you mean :D

**Pull request merged**